### ISSUE TITLE
Suspicious arguments in Switch. "missing break" but has defined exit point

### ISSUE THREAD

**Issue by j-applese3d** [kalessil/phpinspectionsea#1857]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Plugin         | Php Inspections (EA Extended) 4.0.7.1 |
| Language level | PHP 8.0  |

Current behaviour
--
![image](https://user-images.githubusercontent.com/24463871/199076437-72b95f1b-29d5-4527-9396-fca1ebb0f644.png)

Expected behaviour
--
If the switch case is guaranteed to exit, it should not see the "re-declaration" of a variable as a problem.
There are a few ways to define a function as not returning:
```php
// in the function's PHPDoc comment: 
/** @returns never-return[s] */
// in .phpstorm.meta.php: 
exitPoint(\functionName);
// by setting an Attribute for the function: 
#[\JetBrains\PhpStorm\NoReturn]
```
Perhaps all of these should be checked, and if the previous `case` always exits, there is no suspicious argument?

**Comment by jdreesen** [comment]
Since PHP 8.1 there's also the [`never` return type](https://www.php.net/manual/en/language.types.declarations.php#language.types.declarations.never)
